packages/backend-core/src/queue/inMemoryQueue.ts

```diff
@@ -86,10 +87,13 @@ class InMemoryQueue implements Partial<Queue> {
    */
   async process(concurrencyOrFunc: number | any, func?: any) {
     func = typeof concurrencyOrFunc === "number" ? func : concurrencyOrFunc
-    this._emitter.on("message", async message => {
+    this._emitter.on("message", async msg => {
+      const message = cloneDeep(msg)
```

Comment:
Hilariously this is what was causing those `loop.spec.ts` tests to pass without timing out. The orchestrator removes `appId` from the job message, which mutates the original message, causing all future invocations of that message to fail. Preserving the `appId` fixes these, but causes `loop.spec.ts` to then start timing out because it's doing a lot more work than before.